### PR TITLE
Worker flag update callback

### PR DESCRIPTION
**Pull request by chrisfields0**
**Contributions**: We are not currently taking public contributions - see our [contributions](CONTRIBUTING.md) policy. However, we are accepting issues and we do want your [feedback](../README.md#give-us-feedback).

-------

#### Description
Added a delegate that can be binded to in C++ to register for when worker flag updates are received

JIRA ticket: https://improbableio.atlassian.net/projects/UNR/issues/UNR-1605
#### Release note
Added a delegate that can be binded to in C++ to register for when worker flag updates are received which will allow users to do Live Tweaking to values in a cloud deployment

#### Tests
Tested locally by binding to the delegate and manually calling it

#### Documentation
Refer to worker flags documentation on SpatialOS docs for OnFlagsUpdatedCallback

#### Primary reviewers
@m-samiec @joshuahuburn

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDK/Public/Interop/SpatialWorkerFlags.h [hunk 1]
```diff
@@ -5,6 +5,7 @@
 #include "Kismet/BlueprintFunctionLibrary.h"
 #include "SpatialWorkerFlags.generated.h"
 
+DECLARE_DYNAMIC_MULTICAST_DELEGATE_TwoParams(FOnWorkerFlagsUpdated, FString, FlagName, FString, FlagValue);
```
Comment (Vatyx) [inline SpatialGDK/Source/SpatialGDK/Public/Interop/SpatialWorkerFlags.h]:
```suggestion
DECLARE_DYNAMIC_MULTICAST_DELEGATE_TwoParams(FOnWorkerFlagsUpdated, FString, FlagName, FString, FlagValue);

```

File: SpatialGDK/Source/SpatialGDK/Public/Interop/SpatialWorkerFlags.h [hunk 2]
```diff
@@ -19,6 +20,7 @@ class SPATIALGDK_API USpatialWorkerFlags : public UBlueprintFunctionLibrary
 	UFUNCTION(BlueprintCallable, Category="SpatialOS")
 	static bool GetWorkerFlag(const FString& Name, FString& OutValue);
 
+	static FOnWorkerFlagsUpdated OnWorkerFlagsUpdated;
```
Comment (Vatyx) [inline SpatialGDK/Source/SpatialGDK/Public/Interop/SpatialWorkerFlags.h]: Hmmmm, can you make this blueprint exposed?

Comment (chrisfields0) [reply SpatialGDK/Source/SpatialGDK/Public/Interop/SpatialWorkerFlags.h]: No. uproperties can't be static

**Comment by m-samiec** [comment]
@chrisfields0 Was there a Jira ticket for this? And please add a release note also.

File: SpatialGDK/Source/SpatialGDK/Public/Interop/SpatialWorkerFlags.h (hunk 2 above)
Comment (chrisfields0) [reply SpatialGDK/Source/SpatialGDK/Public/Interop/SpatialWorkerFlags.h]: @Vatyx added a function to bind to multicast delegate from BP

File: SpatialGDK/Source/SpatialGDK/Public/Interop/SpatialWorkerFlags.h [hunk 3]
```diff
@@ -19,10 +22,16 @@ class SPATIALGDK_API USpatialWorkerFlags : public UBlueprintFunctionLibrary
...
+	UFUNCTION(BlueprintCallable, Category = "SpatialOS")
+	static void BindToOnWorkerFlagsUpdated(const FOnWorkerFlagsUpdatedBP& InDelegate);
+
+	static FOnWorkerFlagsUpdated OnWorkerFlagsUpdated;
 private:
 	static void ApplyWorkerFlagUpdate(const struct Worker_FlagUpdateOp& Op);
 
 	static TMap<FString, FString> WorkerFlags;
 
+
```
Comment (Vatyx) [inline SpatialGDK/Source/SpatialGDK/Public/Interop/SpatialWorkerFlags.h]: Can you delete these newlines

File: SpatialGDK/Source/SpatialGDK/Public/Interop/SpatialWorkerFlags.h (hunk 2 above)
Comment (Vatyx) [reply SpatialGDK/Source/SpatialGDK/Public/Interop/SpatialWorkerFlags.h]: Can you add something to unbind as well?

Comment (chrisfields0) [reply SpatialGDK/Source/SpatialGDK/Public/Interop/SpatialWorkerFlags.h]: added

File: SpatialGDK/Source/SpatialGDK/Public/Interop/SpatialWorkerFlags.h [hunk 4]
```diff
@@ -19,6 +22,13 @@ class SPATIALGDK_API USpatialWorkerFlags : public UBlueprintFunctionLibrary
 	UFUNCTION(BlueprintCallable, Category="SpatialOS")
 	static bool GetWorkerFlag(const FString& Name, FString& OutValue);
 
+	UFUNCTION(BlueprintCallable, Category = "SpatialOS")
+	static void BindToOnWorkerFlagsUpdated(const FOnWorkerFlagsUpdatedBP& InDelegate);
+
+	UFUNCTION(BlueprintCallable, Category = "SpatialOS")
+	static void UnbindFromOnWorkerFlagsUpdated(const FOnWorkerFlagsUpdatedBP& InDelegate);
+
+	static FOnWorkerFlagsUpdated OnWorkerFlagsUpdated;
```
Comment (Vatyx) [inline SpatialGDK/Source/SpatialGDK/Public/Interop/SpatialWorkerFlags.h]: I think this should be made private

**Pull request merged**